include/aws_iot_mqtt_client_interface.h

```diff
@@ -58,154 +58,242 @@ extern "C" {
 #include "network_interface.h"
 #include "timer_interface.h"
 
+/**
+ * @functionspage{mqtt,MQTT library}
+ *
+ * API functions
+ * - @functionname{mqtt_function_init}
+ * - @functionname{mqtt_function_free}
+ * - @functionname{mqtt_function_connect}
+ * - @functionname{mqtt_function_publish}
+ * - @functionname{mqtt_function_subscribe}
+ * - @functionname{mqtt_function_resubscribe}
+ * - @functionname{mqtt_function_unsubscribe}
+ * - @functionname{mqtt_function_disconnect}
+ * - @functionname{mqtt_function_yield}
+ * - @functionname{mqtt_function_attempt_reconnect}
+ * - @functionname{mqtt_function_get_next_packet_id}
+ * - @functionname{mqtt_function_set_connect_params}
+ * - @functionname{mqtt_function_is_client_connected}
+ * - @functionname{mqtt_function_get_client_state}
+ * - @functionname{mqtt_function_is_autoreconnect_enabled}
+ * - @functionname{mqtt_function_set_disconnect_handler}
+ * - @functionname{mqtt_function_autoreconnect_set_status}
+ * - @functionname{mqtt_function_get_network_disconnected_count}
+ * - @functionname{mqtt_function_reset_network_disconnected_count}
+ */
 
 /**
-* @brief Clean mqtt client from all dynamic memory allocate
-*
-* This function will free up memory that was dynamically allocated for the client.
-*
-* @param pClient MQTT Client that was previously created by calling aws_iot_mqtt_init
-* @return An IoT Error Type defining successful/failed freeing
-*/
- IoT_Error_t aws_iot_mqtt_free( AWS_IoT_Client *pClient );
+ * @functionpage{aws_iot_mqtt_init,mqtt,init}
+ * @functionpage{aws_iot_mqtt_free,mqtt,free}
+ * @functionpage{aws_iot_mqtt_connect,mqtt,connect}
+ * @functionpage{aws_iot_mqtt_publish,mqtt,publish}
+ * @functionpage{aws_iot_mqtt_subscribe,mqtt,subscribe}
+ * @functionpage{aws_iot_mqtt_resubscribe,mqtt,resubscribe}
+ * @functionpage{aws_iot_mqtt_unsubscribe,mqtt,unsubscribe}
+ * @functionpage{aws_iot_mqtt_disconnect,mqtt,disconnect}
+ * @functionpage{aws_iot_mqtt_yield,mqtt,yield}
+ * @functionpage{aws_iot_mqtt_attempt_reconnect,mqtt,attempt_reconnect}
+ */
 
 /**
- * @brief MQTT Client Initialization Function
+ * @brief Initialize a new MQTT client context.
  *
- * Called to initialize the MQTT Client
+ * This function should be called before any other MQTT function to initialize
+ * a new MQTT client context. Once the client context is no longer needed,
+ * @ref mqtt_function_free should be called.
  *
- * @param pClient Reference to the IoT Client
- * @param pInitParams Pointer to MQTT connection parameters
+ * @param[in] pClient MQTT client context to initialize
+ * @param[in] pInitParams Pointer to MQTT connection parameters
```

Comment:
(Nit) ~~Pointer to~~ _The_ Mqtt connection parameters. (Pointer type is expressed in the parameter datatype)

Reply:
fixed